### PR TITLE
chore: add documentation for adding a component library to the creator

### PR DESCRIPTION
**Pull request by janechu**
# Pull Request

## 📖 Description

This change updates the spec file with details on how adding a component library will work in the Creator app.

## 👩‍💻 Reviewer Notes

Read through the change and note any part that does not make sense. Keep in mind the messaging between `creator.tsx` and `preview.tsx` is because the preview exists in an iframe.

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: sites/fast-creator/specs/site.spec.md [hunk 1]
```diff
@@ -19,4 +20,88 @@ Libraries are available in the component tab in the left pane, they can be enabl
 
 Each library will be defined by a configuration which includes the web components themselves, data examples that will be used when a component is added, the tag name of the components and the display name. See the [typings](./app/configs/typings.ts) file for the interface.
 
-When a user adds a library to a project, this library configuration will be dynamically imported.
+When a user adds a library to a project, this library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, this file will be used to create a UI.
```
Comment (nicholasrice) [inline sites/fast-creator/specs/site.spec.md:23]: Maybe it's just I'm missing some context from prior doc sections, but I'm getting a little lost on what "this" is ("this library configuration", "this file"). Should that get re-worded?

Comment (janechu) [inline sites/fast-creator/specs/site.spec.md:23]:
```suggestion
When a user adds a library to a project, the library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, the configuration file will be used to create a UI.
```

Comment (janechu) [reply sites/fast-creator/specs/site.spec.md]: I've clarified by replacing "this" with what it refers to.

Comment (nicholasrice) [inline sites/fast-creator/specs/site.spec.md]:
```suggestion
When a user adds a library to a project, the library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, which will be used to create a UI.
```

Took a stab at what I think you mean that reads a bit more clearly to me. If it's not accurate please disregard

Comment (janechu) [reply sites/fast-creator/specs/site.spec.md]: It's accurate, I just pushed a change that updates it to something similar.

**Review by nicholasrice — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]

File: sites/fast-creator/specs/site.spec.md [hunk 2]
```diff
@@ -19,4 +20,88 @@ Libraries are available in the component tab in the left pane, they can be enabl
 
 Each library will be defined by a configuration which includes the web components themselves, data examples that will be used when a component is added, the tag name of the components and the display name. See the [typings](./app/configs/typings.ts) file for the interface.
 
-When a user adds a library to a project, this library configuration will be dynamically imported.
+When a user adds a library to a project, the library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, the configuration file will be used to create a UI.
```
Comment (janechu) [inline sites/fast-creator/specs/site.spec.md:23]:
```suggestion
When a user adds a library to a project, the library will be dynamically imported. Available libraries will be referenced in a dictionary file, and the library configuration file will be used to create a UI.
```

**Review by EisenbergEffect — APPROVED** [review]

**Pull request merged**